### PR TITLE
Fix SQLAlchemy 1.2.0 compatibility

### LINKED ISSUE #411: Broken compatibility with new release of SQLAlchemy 1.2.0
Hello,

Yesterday released new version of SQLAlchemy (1.2.0) and new release incompatible with aiopg:

```
mymodule.py:42: in fetchone
    result = await conn.execute(query)
.tox/py36-tests/lib/python3.6/site-packages/aiopg/utils.py:72: in __await__
    resp = yield from self._coro
.tox/py36-tests/lib/python3.6/site-packages/aiopg/sa/connection.py:116: in _execute
    return ResultProxy(self, cursor, self._dialect, result_map)
.tox/py36-tests/lib/python3.6/site-packages/aiopg/sa/result.py:234: in __init__
    self._metadata = ResultMetaData(self, cursor.description)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <aiopg.sa.result.ResultMetaData object at 0x7f0417f6a550>
result_proxy = <aiopg.sa.result.ResultProxy object at 0x7f0417f6a5c0>
metadata = (Column(name='column_0', type_code=23, display_size=None, internal_size=4, precision=None, scale=None, null_ok=None), ...Column(name='column_2', type_code=3802, display_size=None, internal_size=-1, precision=None, scale=None, null_ok=None))

    def __init__(self, result_proxy, metadata):
        self._processors = processors = []
    
        result_map = {}
        if result_proxy._result_map:
            result_map = {elem[0]: elem[3] for elem in
                          result_proxy._result_map}
    
        # We do not strictly need to store the processor in the key mapping,
        # though it is faster in the Python version (probably because of the
        # saved attribute lookup self._processors)
        self._keymap = keymap = {}
        self.keys = []
        dialect = result_proxy.dialect
>       typemap = dialect.dbapi_type_map
E       AttributeError: 'PGDialect_psycopg2' object has no attribute 'dbapi_type_map'
```

### PR DESCRIPTION
**Pull request by Gr1N**
Fixes #411

Related SQLAlchemy commit https://github.com/zzzeek/sqlalchemy/commit/31f80b9eaeb3c3435b7f6679b41e434478b1d11c

### CODE REVIEW

**Comment by asvetlov** [comment]
Does the patch work with SQLAlchemy 1.1?

**Comment by Gr1N** [comment]
@asvetlov tests are passing with version 1.1, but as you can see in commit https://github.com/zzzeek/sqlalchemy/commit/31f80b9eaeb3c3435b7f6679b41e434478b1d11c SQLAlchemy bumped minimum required version of `cx_Oracle` and removed `dbapi_type_map` for all engines.

File: aiopg/sa/result.py [hunk 1]
```diff
@@ -121,9 +120,7 @@ def __init__(self, result_proxy, metadata):
             name, obj, type_ = (
                 colname,
                 None,
-                result_map.get(
-                    colname,
-                    typemap.get(coltype, sqltypes.NULLTYPE))
```
Comment (jettify) [inline aiopg/sa/result.py]: We can do `hasattr(dialect, 'dbapi_type_map')` check, and make code working with both version of sqlalchemy

Comment (Gr1N) [reply aiopg/sa/result.py]: if we want to keep `dbapi_type_map` compatibility, I suggest something like:
```
typemap = getattr(dialect, 'dbapi_type_map', {})
```
and keep `result_map` as is.

Comment (Gr1N) [reply aiopg/sa/result.py]: @jettify if you prefer this way I can fix my changes

Comment (jettify) [reply aiopg/sa/result.py]: Looks good, please also add comment explaining why we do that hack

Comment (Gr1N) [reply aiopg/sa/result.py]: Done, please review

**Comment by jettify** [comment]
LGTM, not sure if we we want to test aiopg with both versions of SQLA, change looks trivial. Anyway I will create ticket for to drop this hack and bump minimal required version for SQLA, after one or two releases, this will give some buffer time for upgrade.

**Comment by Gr1N** [comment]
@jettify could you please upload new version of `aiopg` package to PyPI? I really want to upgrade my projects to SQLAlchemy 1.2 :)

**Comment by jettify** [comment]
already deployed https://pypi.python.org/pypi/aiopg/0.13.2

**Comment by Gr1N** [comment]
Thanks!

**Comment by jettify** [comment]
Just noticed other issue: https://github.com/psycopg/psycopg2/issues/594 we need to bump minimal version of psycopg2, as result half of our builds fail due to this bug.

**Pull request merged**